### DISCUSSION TITLE
I want to make drone with mujoco but the problems is the error: more than 6 dofs in body

### DISCUSSION THREAD

**Discussion by jeongsukchul** [google-deepmind/mujoco#987] [Asking for Help]
![image](https://github.com/deepmind/mujoco/assets/124324717/790b2652-964c-454e-a962-91bf80ada64c)
i want to have force and torque generator for that 8 rotors, but in mujoco xml
actuator can only defined by joints. I use slide joint to express force and use hinge joint to express torque.
then, i have to use 16 joints to generate that kinds of force.
I know the problem is because of the (position/velocity) actuator, so one body can't have more than 6 dofs.
Is there any way can I get through this?

**Comment by yuvaltassa** [comment]
> actuator can only defined by joints

What gives you this idea?

You want a free joint with site actuators that apply a force/torque combination at the rotors.

I think @kevinzakka was working on a model?

**Reply by jeongsukchul** [reply]
Thank you for your advice.
https://mujoco.readthedocs.io/en/latest/models.html
https://github.com/deepmind/mujoco_menagerie
Is there appropriate model? It seems like they are all based on hinge joint.. I can't find the free joint you mensioned

**Reply by kevinzakka** [reply]
Have 2 more models I want to add to menagerie (skydio drone + hello stretch). Hopefully I get to them soon :)

**Reply by jeongsukchul** [reply]
Oh, I got it. I'll be looking forward to it! Thanks

**Reply by kevinzakka** [reply]
@jeongsukchul Model is now available [here](https://github.com/google-deepmind/mujoco_menagerie/tree/main/skydio_x2).
